addons/product/models/product_product.py

```diff
@@ -10,6 +10,16 @@
 from odoo.tools import float_compare
 
 
+def _get_barcodes_by_company(products_recordset):
+    barcodes_by_company = defaultdict(list)
+
+    for p in products_recordset:
+        if p.barcode:
+            barcodes_by_company[p.company_id.id].append(p.barcode)
+
+    return barcodes_by_company
```

Comment:
No reasons for this to be outside of the class, move it inside and use `self`.
```suggestion
def _get_barcodes_by_company(self):
    barcodes_by_company = defaultdict(list)

    for product in self:
        if product.barcode:
            barcodes_by_company[product.company_id.id].append(product.barcode)

    return barcodes_by_company
```

Reply:
Moved the method into the class :+1: